### ISSUE TITLE
Validation methods

### ISSUE THREAD

**Issue by jgrewe** [G-Node/nix#141]
We need validation methods/tools that check validity on two levels. 
1. Structural validity on entity level: i.e. are all obligatory fields present, do e.g.  extent and  position match in the number of entries, existence of referenced entities,
2. Logical correctness of connections between entities. This includes checks like do tags point meaningfully into referenced DataArrays etc. 

Feel free to complete the following list.
### DataArray
1. Structural
   
   > _name_, _type_, _unit_, and _label_ are there, The number of _Dimensions_ matches the dimensionality of the data, _sources_ and metadata _section_ exist.
2. Logical
   
   >  Dimension descriptions actually match the data, i.e. The number of tics in a Range dimension match, the duration and sampling interval in a SampledDimension match, the number of labels in SetDimensions match the layout of the data. 
### DataTag
1. Structural
   
   >  _name_, _type_, _unit_, and _label_ are present. _positions_ and _extents_, if present, match in size. _References_, _Representations_, _sources_, and metadata _section_ exist.
2. Logical
   
   > Do positions and extents make sense for all _references_? Same for _Representations_
### SimpleTag

basically same as DataTag
1. Structural

> Number of _units_ matches the number of entries in the _position_ and, if present, the _extent_ vectors.
1. Logical
   > Units stored in the _units_ vector are scalable versions of the units in the referenced _DataArray_(s)

...to be continued!
